modules/client/src/internal/client/methods.ts

```diff
@@ -96,23 +111,55 @@ export class ClientMethods extends ClientCore implements IClientMethods {
       throw new Error("Could not pin the metadata on IPFS");
     }
 
-    // @ts-ignore  TODO: Remove this comment when used
-    const registryAddress = await daoFactoryInstance.registry();
+    const tx = await daoFactoryInstance.connect(signer).createDao(
+      {
+        name: params.ensSubdomain,
+        metadata: toUtf8Bytes(cid),
+        trustedForwarder: params.trustedForwarder || AddressZero,
+      },
+      pluginInstallationData
+    );
 
-    // TODO: Remove mock result
-    await delay(1000);
     yield {
       key: DaoCreationSteps.CREATING,
-      txHash:
-        "0x1298376517236498176239851762938512359817623985761239486128937461",
+      txHash: tx.hash,
     };
+    // get dao registry address
+    const daoRegistryAddress = await daoFactoryInstance.daoRegistry();
+    // start tx
+    const receipt = await tx.wait();
+    // find dao address using the dao registry address
+    const newDaoAddress = receipt.events?.find(
+      e => e.address === daoRegistryAddress
```

Comment:
Isn't it possible to filter by the event name? this should do the trick too... unless there ever is a second event there